mitodl/mitxonline
# 806: ecommerce implement a cybersource notification api endpoint

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/806

#### What's this PR do?
Adds an additional API endpoint that Cybersource will send a POST request to in response to a user's payment submission.  This PR does a lot of refactoring within the project to separate much of the validation and processing of the Cybersource request so that it can be used by the new API endpoint `checkout_result_api` and `checkout-result-callback`.  

The `checkout-result-callback` is where the user is redirected to after completing the payment submission on Cybersource, and so this view contains extra functionality to show the user a message based on the payment status.

The `checkout_result_api` will always respond with a 200, unless the Order status is `PENDING` and the Cybersource signature on the request could not be validated.

#### How should this be manually tested?
I spent a lot of time building the unit tests to ensure that they were able to run through as much functionality as possible.  I also made sure that I could hit the endpoint locally using Postman.  I'm not sure what's the best way to test this locally though.
